molecule/vagrant_packager/library/molecule_vagrant.py

```diff
@@ -0,0 +1,456 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#  Copyright (c) 2015-2017 Cisco Systems, Inc.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
```

Comment:
This is a gigantic file, and it was introduced as part of a beefy commit with no explanation about _why_ it's here:

```
$ git show 542c2cbd2e6d27e0bcd0f58cb472cb7dda9ce3f1 --name-status
commit 542c2cbd2e6d27e0bcd0f58cb472cb7dda9ce3f1
Author: Michael Sheinberg <mike@freedom.press>
Date:   Mon Feb 26 16:42:28 2018 -0500

    Add scenario for vagrant builder

M       .gitignore
A       molecule/vagrant_packager/ansible-override-vars.yml
A       molecule/vagrant_packager/box_files/Vagrantfile.app-staging
A       molecule/vagrant_packager/box_files/Vagrantfile.mon-staging
A       molecule/vagrant_packager/box_files/vagrant_app-staging.yml
A       molecule/vagrant_packager/box_files/vagrant_mon-staging.yml
A       molecule/vagrant_packager/build/.gitignore
A       molecule/vagrant_packager/create.yml
A       molecule/vagrant_packager/destroy.yml
A       molecule/vagrant_packager/library/molecule_vagrant.py
A       molecule/vagrant_packager/molecule.yml
A       molecule/vagrant_packager/package.py
A       molecule/vagrant_packager/playbook.yml
A       molecule/vagrant_packager/prepare.yml
A       molecule/vagrant_packager/reboot_and_wait.yml
A       molecule/vagrant_packager/sd_clone.yml
A       molecule/vagrant_packager/side_effect.yml
```

I assume it contains a patch of some kind—perhaps already included in upstream?—but at the very least the commit message should explain the motivation to include, and ideally also parameters that would suffice for removal later.

Reply:
got an incoming commit to address this -- you are right i should have been more explicit though about why it was needed. long story short, upstream was missing functionality so i was waiting for my changes and others to get pulled in